node/overseer/src/tests.rs

```diff
@@ -1129,7 +1129,9 @@ fn context_holds_onto_message_until_enough_signals_received() {
 
 	let mut ctx = OverseerSubsystemContext::new(
 		signal_rx,
-		stream::select(bounded_rx, unbounded_rx),
+		stream::select_with_strategy(bounded_rx, unbounded_rx, |_: &mut ()| -> stream::PollNext {
+			stream::PollNext::Right
```

Comment:
Hmm, under high load cases, we'd starve the bounded channel and probably cause other problems

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=80495ec5912019ccbee6adc99a869966

I'd rather consider something that only biases the processing towards the unbounded, but that's not based on anything and makes it harder to reason about the system.

Mostly some food for thought.

Reply:
The reasons for that are the following:

* unbounded channels can grow indefinitely, so they have to be processed as soon as possible to avoid that
* we mostly use unbounded channels as high priority channels but with the round-robin polling strategy it is not guaranteed
* if an unbounded channel is empty, there is no real penalty for the bounded channel
* we have several magnitudes less unbounded messages than bounded messages according to the grafana board